test/unit/app/controllers/detect-tokens-test.js

```diff
@@ -0,0 +1,93 @@
+const assert = require('assert')
+const sinon = require('sinon')
+const DetectTokensController = require('../../../../app/scripts/controllers/detect-tokens')
+const PreferencesController = require('../../../../app/scripts/controllers/preferences')
+const ObservableStore = require('obs-store')
+
+describe('DetectTokensController', () => {
+  it('should poll on correct interval', async () => {
+    const stub = sinon.stub(global, 'setInterval')
+    new DetectTokensController({ interval: 1337 }) // eslint-disable-line no-new
+    assert.strictEqual(stub.getCall(0).args[1], 1337)
+    stub.restore()
+  })
+
+  it('should not check tokens while in test network', async () => {
+    var network = new ObservableStore({provider: {type: 'rinkeby'}})
+    const preferences = new PreferencesController()
+    const controller = new DetectTokensController({preferences: preferences, network: network})
+    controller.isActive = true
+    controller.contracts = {
+        '0x0D262e5dC4A06a0F1c90cE79C7a60C09DfC884E4': {
+            'name': 'JET8 Token',
+            'logo': 'J8T.svg',
+            'erc20': true,
+            'symbol': 'J8T',
+            'decimals': 8,
+        },
+        '0xBC86727E770de68B1060C91f6BB6945c73e10388': {
+            'name': 'Ink Protocol',
+            'logo': 'ink_protocol.svg',
+            'erc20': true,
+            'symbol': 'XNK',
+            'decimals': 18,
+        },
+      }
+    controller.fetchContractAccountBalance = address => address
```

Comment:
Might be worth using `sinon.stub` here, so it can be automatically cleaned up after the test.